src/ppom-controller.ts

```diff
@@ -239,9 +243,33 @@ export class PPOMController extends BaseControllerV2<
     });
     this.#ppomMutex = new Mutex();
 
-    onNetworkChange((id: string) => {
+    onNetworkChange((networkControllerState: any) => {
+      const id = networkControllerState.providerConfig.chainId;
+      if (id === this.state.chainId) {
+        return;
+      }
+      let { chainIdCache } = this.state;
+      const existingNetworkObject = chainIdCache.find(
+        ({ chainId: cid }) => cid === id,
+      );
+      if (existingNetworkObject) {
```

Comment:
Both conditions here are very similar except for the `dataFetched` value and whether we remove the old value. If the cache becomes an object, we don't have to worry about removal so we could merge these conditions and just conditionally set `dataFetched` based on `existingNetworkObject`?